src/main/java/com/getindata/connectors/http/internal/utils/uri/URIBuilder.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * ====================================================================
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * ====================================================================
+ *
+ * This software consists of voluntary contributions made by many
+ * individuals on behalf of the Apache Software Foundation.  For more
+ * information on the Apache Software Foundation, please see
+ * <http://www.apache.org/>.
+ *
+ * ============================= NOTE =================================
+ * This code has been copied from
+ * https://github.com/apache/httpcomponents-client/tree/rel/v4.5.13
+ * and it was changed to use in this project.
+ * ====================================================================
+ */
+
+package com.getindata.connectors.http.internal.utils.uri;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.flink.util.StringUtils;
+
+/**
+ * Builder for {@link URI} instances.
+ */
```

Comment:
JavaDocs for public-facing methods and fields should be mentioning that "it is based of <link-to-original-apache-implementation>"

Reply:
fixed